From f70e0e28f44f01233d6804ebef815af9a211c66d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 22:20:02 +0800
Subject: [PATCH 31/35] [r38460] ae_lavc: Generate a guessed AVChannelLayout
 based on the number of channels

Otherwise most (all?) FFmpeg encoders won't work.

(From upstream SVN r38460)
---
 libmpcodecs/ae_lavc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libmpcodecs/ae_lavc.c b/libmpcodecs/ae_lavc.c
index 39a0d28..a9f49d0 100644
--- a/libmpcodecs/ae_lavc.c
+++ b/libmpcodecs/ae_lavc.c
@@ -192,7 +192,7 @@ int mpae_init_lavc(audio_encoder_t *encoder)
             mp_msg(MSGT_MENCODER,MSGL_ERR, "Audio encoder requires unknown or unsupported input format\n");
             return 0;
 	}
-	lavc_actx->ch_layout.nb_channels = encoder->params.channels;
+	av_channel_layout_default(&lavc_actx->ch_layout, encoder->params.channels);
 	lavc_actx->sample_rate = encoder->params.sample_rate;
 	lavc_actx->time_base.num = 1;
 	lavc_actx->time_base.den = encoder->params.sample_rate;
-- 
2.46.0

